File: a2q3.py
#importing the text file in the same directory as 
filename = 'Mmatrix.txt'

#creating empty list and strings for each line
R1=[]
R2=[]
R3=[]
string1 = ''
string2 = ''
string3 = ''
with open(filename) as file_object:
	#reading the lines
	lines = file_object.readlines()


#For first column
for line in lines[0]:
	#stripping all spaces
	string1 += line.strip()
	#converting strings into list

	f'{list(string1)}'


for i in string1:
	R1.append(int(i))
#print(R1)

#for second column
for line in lines[1]:
	string2 += line.strip()

	f'{list(string2)}'


for i in string2:
	R2.append(int(i))
#print(R2)

#for 3rd column
for line in lines[2]:
	string3 += line.strip()

	f'{list(string3)}'
	

for i in string3:
	R3.append(int(i))
#print(R3)
#matrix 3X3 form
M =[R1,
	R2,
	R3]

print(f'M = {M}')

#using similar operation over new variables or lists
filename = 'Nmatrix.txt'


R1=[]
R2=[]
R3=[]
string1 = ''
string2 = ''
string3 = ''
with open(filename) as file_object:
	lines = file_object.readlines()



for line in lines[0]:
	string1 += line.strip()

	f'{list(string1)}'
	#ok = print(f'{list(M.strip)}')

for i in string1:
	R1.append(int(i))
#print(R1)

for line in lines[1]:
	string2 += line.strip()

	f'{list(string2)}'
	

for i in string2:
	R2.append(int(i))
#print(R2)

for line in lines[2]:
	string3 += line.strip()

	f'{list(string3)}'


for i in string3:
	R3.append(int(i))
#print(R3)

N =[R1,
	R2,
	R3]

print(f'N= {N}')

lol=[]
McrossN =[]
#creating empty lists for each column
c1=[]
c2=[]
c3=[]
#looping over index of outer list
for i in range(0,3):
	#looping over index of inner list for bth M and N
	for j in range(0,3):
		#defining cross and adding each time j is looped
		cross = 0
		for k in range(0,3):	
			cross += M[i][k]*N[k][j]
#As we need to add after looping through each 			
		lol.append(cross)
#breaking list as per required column
c1 = lol[0:3]
c2 = lol[3:6]
c3 = lol[6:9]
#defining M X N
McrossN = [ c1,
			c2,
			c3]
print(f'M X N = {McrossN}')

#for MXA
A = [1,2,3]

lol2 =[]

for i in range(0,3):
	cross2 = 0
	for k in range(0,3):
		cross2 += M[i][k]*A[0]
	lol2.append(cross2)
print(f'M X A = {lol2}')	


#OUTPUT
#M = [[1, 2, 3], [4, 5, 6], [7, 8, 9]]
#N= [[1, 1, 2], [3, 1, 5], [6, 1, 8]]
#M X N = [[25, 6, 36], [55, 15, 81], [85, 24, 126]]
#M X A = [6, 15, 24]

File: a2q3.c
#include <stdio.h>
#include <stdlib.h>

int main()
{
//define variables
    int sum = 0;
    int i,j;
    double value[3][3];
    double va[3][3];
    double A[3][1];
    double mul[3][1];
    double multiply[3][3];

//define A
    A[0][0]=2;
    A[1][0]=3 ;
    A[2][0]=-5 ;
 //open file in directory   
    FILE *open;
    FILE *opena;
    open = fopen("Mmatrix.txt","r");
    
    i=0;
    while (feof(open) == 0)
    {
        fscanf( open, "%lf %lf %lf", &value[i][0],&value[i][1],&value[i][2]);
        i++;
        
    }
    
   
    fclose(open);
    
    opena = fopen("Nmatrix.txt","r");
    j=0;

    
    while (feof(opena) ==0)
    {
        fscanf( opena, "%lf %lf %lf", &va[j][0],&va[j][1],&va[j][2]);
        j++;
        
    }

    fclose(opena);
    
    for (int c = 0; c < 3; c++) {
      for (int d = 0; d < 3; d++) {
        
           multiply[c][d]=0 ;
        for (int k = 0; k < 3; k++) {
 
        multiply[c][d] += value[c][k]*va[k][d];
        }
      }
    }
    
    for (int c = 0; c < 3; c++) {
      for (int d = 0; d <1; d++) {
        
           mul[c][d]=0 ;
        for (int k = 0; k < 3; k++) {
 
        mul[c][d] += value[c][k]*A[k][d];
        }
      }
    }
    
     printf(" A = \n");
    for(j = 0; j < 3; j++)
        printf("%10.0f \n", A[j][0]);
   
    printf(" M =\n");
    for(j = 0; j < i; j++)
        printf("%10.0f %10.3f %10.3f\n", value[j][0], value[j][1], value[j][2]);
        
    printf(" N = \n");
    for(j = 0; j < i; j++)
        printf("%10.0f %10.3f %10.3f\n", va[j][0], va[j][1], va[j][2]);
    
    printf(" M x N =\n");
    for(j = 0; j < i; j++)
        printf("%10.0f %10.3f %10.3f\n", multiply[j][0], multiply[j][1], multiply[j][2]);
        
    printf(" M X A = \n");
    for(j = 0; j < i; j++)
        printf("%10.0f \n", mul[j][0]);
    
    
    
    return 0;
}

//OUTPUT
// M =
//1 2 3
//4 5 6
//7 8 9


// N =
//1 1 2
//3 1 5
//6 1 8


// M x N =
// 25, 6, 36                                                                                                          
// 55, 15, 81                                                                                                           
// 85, 24, 126

// M x A =
// 6                                                                                                                                   
// 15                                                                                                                                    
// 28